tools/crm_resource.c

```diff
@@ -802,50 +802,7 @@ main(int argc, char **argv)
         rc = cli_resource_print(rsc_id, &data_set, FALSE);
 
     } else if(rsc_cmd == 'Y') {
-        resource_t *rsc = NULL;
-	GListPtr hosts = NULL;
-        GListPtr lpc = NULL;
-	if (rsc_id == NULL && host_uname == NULL) {
-            for (lpc = data_set.resources; lpc != NULL; lpc = lpc->next) {
-                resource_t *rsc = (resource_t *) lpc->data;
-                rsc->fns->location(rsc, &hosts, TRUE);
-		if ( hosts == NULL ) {
-		    crm_trace("Resource is not running");
-                    cli_resource_check(cib_conn, rsc);
-		} else {
-		    crm_trace("Resource is running");
-		}
-		g_list_free(hosts);
-	        hosts = NULL;
-	    }
-        } else if (rsc_id != NULL && host_uname != NULL) {
-            rsc = pe_find_resource(data_set.resources, rsc_id);
-	    if(!(resource_is_running_on(rsc,host_uname))) {
-                cli_resource_check(cib_conn, rsc);
-            }
-         } else if (rsc_id == NULL && host_uname != NULL) {
-            GListPtr allResources = get_all_resources(host_uname,data_set.resources);
-	    GListPtr activeResources = get_active_resources(host_uname,allResources);
-	    GListPtr unactiveResources = subtract_lists(allResources,activeResources);
-	    for(lpc = unactiveResources; lpc != NULL; lpc = lpc->next) {
-	        resource_t *rsc = (resource_t *) lpc->data;
-	        cli_resource_check(cib_conn,rsc);
-            }
-            g_list_free(allResources);
-            g_list_free(activeResources);
-	    g_list_free(unactiveResources);
-         } else if (rsc_id != NULL && host_uname == NULL) {
-	    rsc = pe_find_resource(data_set.resources, rsc_id);
-            rsc->fns->location(rsc, &hosts, TRUE);
-            if ( hosts == NULL ) {
-	        crm_trace("Resource is not running");
-                cli_resource_check(cib_conn, rsc);
-	    } else {
-		crm_trace("Resource is running");
-            }
-            g_list_free(hosts);
-            hosts = NULL;
-	 } 
+      cli_resource_why(cib_conn,data_set.resources,rsc_id,host_uname);
```

Comment:
This looks good.

Be sure to go through your changes and make sure all the indenting is correct: no tabs, and four spaces per indent.